src/protocol/reveal/mod.rs

```diff
@@ -177,9 +177,9 @@ mod tests {
 
             let record_id = RecordId::from(i);
             let results = try_join_all(vec![
-                ctx[0].clone().reveal(record_id, &share[0]),
-                ctx[1].clone().reveal(record_id, &share[1]),
-                ctx[2].clone().reveal(record_id, &share[2]),
+                ctx[0].ctx.clone().reveal(record_id, &share[0]),
+                ctx[1].ctx.clone().reveal(record_id, &share[1]),
+                ctx[2].ctx.clone().reveal(record_id, &share[2]),
```

Comment:
What is going on here? Why did you need to add a `.ctx`? And what does that even do?

Reply:
make malicious context returns both context and validator (so that it is not dropped). Thats why we need to explicitly call ctx here